### PR TITLE
Clean up feature level requirements to only contain content we want to require

### PR DESCRIPTION
**Pull request by brian-avery**
When discussing the feature level requirements from the definition of done, we found that there was quite a bit of controversy over these requirements for features of various levels. I've opened this PR to come to consensus for these requirements before we integrate them into the pull request in enhancements. After this is approved/merged, these requirements will be merged into the checklist on istio/enhancements so that it's easier to track feature promotion.

### CODE REVIEW

File: FEATURE-LIFECYCLE-CHECKLIST.md [hunk 1]
```diff
@@ -17,25 +17,23 @@ A development feature can be officially labeled as Alpha once it meets the follo
...
 
 * Tests
-    * Test coverage is >= 70%.
-    * Integration tests cover core use cases with the feature enabled.
+    * Automated integration tests cover core use cases with the feature enabled.
     * When disabled the feature does not affect system stability or performance.
 
 * Performance
-    * Performance requirements assessed as part of design.
+    * Performance requirements have been measured.
```
Comment (nrjpoddar) [inline FEATURE-LIFECYCLE-CHECKLIST.md:33]:
```suggestion
    * Performance impacts have been measured.
```

**Review by nrjpoddar — COMMENTED** [review]
@brian-avery should we add some debugging documentation/tooling if applicable for Stable/Beta.

I'm thinking that's how we can leverage the work UX WG is doing and ensure istioctl tool is well maintained for new features.

@therealmitchconnors @esnible WDYT?

**Comment by brian-avery** [comment]
@nrjpoddar that sounds like a great idea to me. I think that should be part of the supportability review. Lets cover that in the enhancements PR though. This PR is more to clarify what stands/remove what we need to.

**Comment by brian-avery** [comment]
@howardjohn @costinm @smawson @linsun This PR exists to clean up the prior feature lifecycle checklist before we merge it with the definition of done. Can you please take a look?

**Review by howardjohn — APPROVED** [review]

File: FEATURE-LIFECYCLE-CHECKLIST.md [hunk 2]
```diff
@@ -74,11 +71,10 @@ An Alpha feature can be officially labeled as Beta once it meets the following a
...
 * Tests
-    * Test coverage is >= 90%
     * Automated tests are in place to prevent performance regressions.
 
 * Performance
-    * Latency, throughput and scale are quantified and documented.
+    * Latency, throughput and scalability are quantified and documented on istio.io.
 
 * Bugs
     * Feature has no outstanding P0 or P1 bugs.
```
Comment (howardjohn) [inline FEATURE-LIFECYCLE-CHECKLIST.md:80]: Sorry my original approval was that this was an incremental improvement not that I now agree with the whole doc. This doesn't make sense IMO. First of all, we have a lot of P1 bugs, that is quite a high bar. And what happens if a bug gets opened 1 day before we promote to stable, or 1 day after? Is it demoted?

Comment (brian-avery) [reply FEATURE-LIFECYCLE-CHECKLIST.md:80]: Title was poorly worded before. I cleaned it up. 

My thought is that there needs to be some sort of a bar. P0/P1 priority is generally applied to a release basis, and I think it makes sense to say that a beta/stable feature should not ship as part of a release if it has issues that the working group leads have marked as a p0 for that release. However, that's on a release level. 

I think there should be something that says that major issues are not present in that feature before it gets promoted though. We could say something along the lines of "Has been approved by appropriate workgroup and that group considers there to be no major issues at the time of promotion". That would leave the onus on the work group to decide if they consider it to be stable (functional without known issues?)

Comment (howardjohn) [reply FEATURE-LIFECYCLE-CHECKLIST.md:80]: That makes sense to me.

Comment (brian-avery) [inline FEATURE-LIFECYCLE-CHECKLIST.md:80]:
```suggestion
    * Feature has no major known issues.
```

File: FEATURE-LIFECYCLE-CHECKLIST.md [hunk 3]
```diff
@@ -64,7 +61,7 @@ An Alpha feature can be officially labeled as Beta once it meets the following a
     * API has had a thorough API review and is thought to be complete.
 
 * CLI
-    * Necessary CLI commands have been implemented and are complete.
+    * Any necessary CLI commands have been implemented and are complete.
 
 * Bugs
     * Feature has no outstanding P0 bugs.
```
Comment (brian-avery) [inline FEATURE-LIFECYCLE-CHECKLIST.md:67]:
```suggestion
    * Feature has no major known issues
```

**Review by linsun — APPROVED** [review]
LGTM

**Comment by linsun** [comment]
The comment from @nrjpoddar is good, and should be considered as part of the re-definition in https://github.com/istio/enhancements/pull/3

The goal of this PR is just to clean up what we have but details will be in https://github.com/istio/enhancements/pull/3.

File: FEATURE-LIFECYCLE-CHECKLIST.md [hunk 4]
```diff
@@ -17,25 +17,23 @@ A development feature can be officially labeled as Alpha once it meets the follo
 requirements:
 
 * Design
-    * Requires Approved RFC in order to merge.
+    * Requires Approved RFC.
 * Config
     * Requires explicit user action to enable (e.g. a config field, config resource, or installation action).
 
 * Docs
-    * Reference docs are published on istio.io.
```
Comment (dom589) [inline FEATURE-LIFECYCLE-CHECKLIST.md:25]: Clear

File: FEATURE-LIFECYCLE-CHECKLIST.md [hunk 5]
```diff
@@ -17,25 +17,23 @@ A development feature can be officially labeled as Alpha once it meets the follo
...
+    * Automated integration tests cover core use cases with the feature enabled.
     * When disabled the feature does not affect system stability or performance.
 
 * Performance
-    * Performance requirements assessed as part of design.
+    * Performance impacts have been measured.
 
 * API
-    * (Optional) Initial API review.
+    * Initial API review.
```
Comment (dom589) [inline FEATURE-LIFECYCLE-CHECKLIST.md:36]: -

File: FEATURE-LIFECYCLE-CHECKLIST.md [hunk 6]
```diff
@@ -17,25 +17,23 @@ A development feature can be officially labeled as Alpha once it meets the follo
 requirements:
 
 * Design
```
Comment (dom589) [inline FEATURE-LIFECYCLE-CHECKLIST.md:19]: -

**Pull request merged**